core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1086,6 +1108,66 @@ class ParserTest extends munit.ScalaCheckSuite {
     }
   }
 
+  property("rep parses at most max entries") {
+    val validMinMax = for {
+      min <- Gen.choose(1, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
+        genP.fa.rep0(min, max).parse(str).foreach { case (_, l) =>
+          assert(l.length <= max)
+        }
+      }
+    }
+  }
+
+  property("rep0 parses at most max entries") {
+    val validMinMax = for {
+      min <- Gen.choose(1, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
+        genP.fa.rep0(min, max).parse(str).foreach { case (_, l) =>
+          assert(l.length <= max)
+        }
+      }
+    }
+  }
+
+  object Counter extends Accumulator0[Unit, Int] {
+    class Counter extends Appender[Unit, Int] {
+      var n = 0
+      def append(item: Unit) = {
+        n += 1
+        this
+      }
+      def finish(): Int = n
+    }
+    def newAppender(): Appender[Unit, Int] = new Counter()
+  }
+
+  property("repAs parses max entries when available") {
+    forAll(Gen.choose(1, 100)) { (n: Int) =>
+      val toBeConsumed = "a" * n
+      val p = Parser.char('a').repAs(min = n, max = n)(Counter)
+      val parsed = p.parseAll(toBeConsumed)
+      assertEquals(parsed, Right(n))
+    }
+  }
+
+  property("repAs parses max entries when more is available") {
+    forAll(Gen.choose(1, 100)) { (n: Int) =>
+      val toBeConsumed = "a" * n * 2
+      val p = Parser.char('a').repAs(min = n, max = n)(Counter)
+      val parsed = p.parse(toBeConsumed)
+      assertEquals(parsed, Right(("a" * n, n)))
+    }
+  }
```

Comment:
Can we add a test that we parse exactly N when min = N and max = N?

Reply:
Do the "rep0/rep/repAs0/repAs parses max entries when available" tests cover this?

Reply:
I think in principle, yes, but as you say, we might not hit it very often (remember, with the scala check, most parsers will just fail to parse anyway, add repetition in there and maybe they almost never parse, so I think the tests may be much weaker than they appear at first glance).

Reply:
Looking at the new tests, I think these are good.